lib/ansible/modules/windows/win_find.ps1

```diff
@@ -250,24 +250,28 @@ Function Get-FileStat($file) {
         try {
             $lnk_source = [Ansible.Command.SymLinkHelper]::GetSymbolicLinkTarget($file)
             $file_stat.lnk_source = $lnk_source
-        } catch {}
+        } catch {
+            #Add-Warning -obj $result -message "symlink source retrieval error: $($_.Exception.Message)"
+        }
     } elseif ($file.PSIsContainer) {
         $isdir = $true
 
-        $share_info = Get-WmiObject -Class Win32_Share -Filter "Path='$($file.Fullname -replace '\\', '\\')'"
+        $share_info = Get-CIMInstance -Class Win32_Share -Filter "Path='$($file.Fullname -replace '\\', '\\')'"
         if ($null -ne $share_info) {
             $isshared = $true
             $file_stat.sharename = $share_info.Name
         }
-        
+
         # only get the size of a directory if there are files (not directories) inside the folder
         # Get-ChildItem -LiteralPath does not work properly on older OS', use .NET instead
         $dir_files = @()
         try {
             $dir_files = $file.EnumerateFiles("*", [System.IO.SearchOption]::AllDirectories)
         } catch [System.IO.DirectoryNotFoundException] { # Broken ReparsePoint/Symlink, cannot enumerate
-        } catch [System.UnauthorizedAccessException] {}  # No ListDirectory permissions, Get-ChildItem ignored this
-
+            Add-Warning -obj $result -message "symlink enumeration error: $($_.Exception.Message)"
+      	} catch [System.UnauthorizedAccessException] {  # No ListDirectory permissions, Get-ChildItem ignored this
+       	    Add-Warning -obj $result -message "symlink enumeration error: $($_.Exception.Message)"
```

Comment:
I would leave this as is for now. We need to look more into win_find and how it enumerates but prefer to do that sometime in the future without adding more warnings for now.I would leave this as is for now. We need to look more into win_find and how it enumerates but prefer to do that sometime in the future without adding more warnings for now.